omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/TaskActionResource1_8.java

```diff
@@ -0,0 +1,220 @@
+public class TaskActionResource1_8 extends BaseDelegatingResource<TaskAction> implements Creatable {
+	
+	private TaskServiceWrapper taskServiceWrapper = new TaskServiceWrapper();
+	
+	public void setTaskServiceWrapper(TaskServiceWrapper taskServiceWrapper) {
+		this.taskServiceWrapper = taskServiceWrapper;
+	}
+	
+	@Override
+	public Object create(SimpleObject post, RequestContext context) throws ResponseException {
+		TaskAction action = newDelegate();
+		setConvertedProperties(action, post, getCreatableProperties(), true);
+		
+		Collection<TaskDefinition> taskDefinitions;
+		if (action.isAllTasks() != null && action.isAllTasks()) {
+			taskDefinitions = taskServiceWrapper.getRegisteredTasks();
+			action.setTasks(new ArrayList<TaskDefinition>(taskDefinitions));
+		} else {
+			taskDefinitions = action.getTasks();
+		}
+		ServletContext servletContext = getServletContext(context);
+		
+		if (action.getAction() == TaskAction.Action.SCHEDULETASK || action.getAction() == TaskAction.Action.RESCHEDULETASK
+		        || action.getAction() == TaskAction.Action.DELETE || action.getAction() == TaskAction.Action.SHUTDOWNTASK) {
+			if (taskDefinitions == null || taskDefinitions.isEmpty()) {
+				throw new IllegalRequestException("Cannot execute action " + action.getAction()
+				        + " on empty set of task definitions.");
+			}
+		}
+		if (action.isAllTasks() == null || !action.isAllTasks()) {
+			for (TaskDefinition taskDef : taskDefinitions) {
+				if (taskDef == null) {
+					throw new IllegalRequestException(
+					        "One or more of the task definitions you specified are not registered on this server");
+				}
+			}
+		}
+		
+		switch (action.getAction()) {
+			case SCHEDULETASK:
+				scheduleTask(taskDefinitions);
+				break;
+			case RESCHEDULETASK:
+				reScheduleTask(taskDefinitions);
+				break;
+			case DELETE:
+				deleteTask(taskDefinitions);
+				break;
+			case SHUTDOWNTASK:
+				shutDownTask(taskDefinitions);
+				break;
+			case RESCHEDULEALLTASK:
+				reScheduleAllTask();
+				break;
+			case ONSTARTUP:
+				onStartUpTask();
+				break;
+			case ONSHUTDOWN:
+				onShutDownUpTask();
+				break;
+		}
+		return ConversionUtil.convertToRepresentation(action, Representation.DEFAULT);
+	}
+	
+	private void scheduleTask(Collection<TaskDefinition> taskDefs) {
+		for (TaskDefinition taskDef : taskDefs) {
+			try {
+				taskServiceWrapper.scheduleTask(taskDef);
+			}
+			catch (SchedulerException e) {
+				throw new APIException("Errors occurred while scheduling task", e);
+			}
+		}
+	}
+	
+	private void shutDownTask(Collection<TaskDefinition> taskDefs) {
+		for (TaskDefinition taskDef : taskDefs) {
+			try {
+				taskServiceWrapper.shutDownTask(taskDef);
+			}
+			catch (SchedulerException e) {
+				throw new APIException("Errors occurred while shutdowning task", e);
+			}
+		}
+	}
+	
+	// Stop and start a scheduled task.
+	private void reScheduleTask(Collection<TaskDefinition> taskDefs) {
```

Comment:
Since this is a collection of taskDefs. Can you add an s and change to reScheduleTasks?
The same applies to the other methods.

Reply:
Sure :-)